### PR TITLE
Fix a number of bugs and issues with scroll roots and transforms.

### PR DESCRIPTION
**Pull request by glennw**
This change is [Reviewable](https://reviewable.io/reviews/servo/webrender/355)

### CODE REVIEW

**Comment by glennw** [comment]
r? @pcwalton 

You have already reviewed the first commit in this patch set. I'm including it here (and closing the PR for that patch) because bors is not working at the moment and won't merge that PR. So only the scroll roots commit in this PR needs to be reviewed.

Note that this also requires a small patch in servo (https://github.com/servo/servo/pull/12932), so WR shouldn't be updated in Servo without that patch.

File: webrender/src/frame.rs [hunk 1]
```diff
@@ -673,203 +467,313 @@ impl Frame {
...
+                                          .get(&pipeline_id)
+                                          .expect("No auxiliary lists?!");
+                stacking_context.composition_operations(auxiliary_lists)
+            };
+
+            // Detect composition operations that will make us invisible.
+            for composition_operation in &composition_operations {
+                match *composition_operation {
+                    CompositionOp::Filter(LowLevelFilterOp::Opacity(Au(0))) => return,
+                    _ => {}
```
Comment (pcwalton) [inline webrender/src/frame.rs]: nit: Could be `if let` I guess

File: webrender/src/frame.rs [hunk 2]
```diff
@@ -673,203 +467,313 @@ impl Frame {
...
-            match (stacking_context.scroll_policy, stacking_context.scroll_layer_id) {
-                (ScrollPolicy::Scrollable, Some(ScrollLayerId {
-                    info: ScrollLayerInfo::Scrollable(index),
-                    ..
-                })) if index > 0 => {
-                    Some(stacking_context.transform
-                                         .invert()
-                                         .transform_rect(&stacking_context.overflow))
+        let scene_items = scene_item.collect_scene_items(&context.scene);
+        if !scene_items.is_empty() {
```
Comment (pcwalton) [inline webrender/src/frame.rs]: Can this be `if scene_items.is_empty() { return }`?

**Comment by pcwalton** [comment]
Looks good, r=me

**Pull request merged**